src/api/app/models/project.rb

```diff
@@ -1645,6 +1616,27 @@ def release_targets_ng
     return release_targets_ng
   end
 
+  # Go through all enabled build flags and look for a repo name that matches a
+  # previously parsed release target name (from "release_targets_ng").
+  #
+  # If one was found return the project name, otherwise return nil.
+  def guess_release_target_from_package(package, parsed_targets)
+    # Stone cold map'o'rama of package.$SOMETHING with package/build/enable/@repository=$ANOTHERTHING to
+    # project/repository/releasetarget/@project=$YETSOMETINGDIFFERENT. Piece o' cake, eh?
+    target_mapping = {}
+    parsed_targets.each do |rt_key, rt_value|
+      target_mapping[rt_value[:reponame]] = rt_key
+    end
+
+    package.flags.where(flag: :build, status: 'enable').each do |flag|
+      rt_key = target_mapping[flag.repo]
+      return rt_key if rt_key
+    end
+
+    nil
+  end
+  private :guess_release_target_from_package
```

Comment:
I don't like 
``
private :guess_release_target_from_package
``

Don't we have already a private section in the project model?